### PR TITLE
Node service

### PR DESCRIPTION
**Pull request by Liang1XZhao**
add node service dockerfile

### CODE REVIEW

File: ehsm-service/index.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+/**
+ * 校验payload
+ */
+ const _checkPayload = function (req,res,next){
+   const action = req.query.Action;
+  const { payload } = req.body;
+  const currentPayLoad = pramas[action];
+  for (const key in currentPayLoad) {
+    if(payload[key] == undefined) {
+      res.send(result(401,'payload参数不全'));
```
Comment (yang8621) [inline ehsm-service/index.js:57]: Change all wordings to English, including all /*comments*/.

File: core/App/ehsm_provider.cpp [hunk 2]
```diff
@@ -553,4 +553,4 @@ EH_RV ExportDataKey(EH_MECHANISM_PTR pMechanism,
         return EHR_OK;
 }
 
-}
```
Comment (yang8621) [inline core/App/ehsm_provider.cpp:556]: Discard this change: 100644 → 100755

File: core/Makefile [hunk 3]
```diff
@@ -164,4 +164,4 @@ $(Enclave_Name): Enclave/enclave_hsm_t.o $(Enclave_Cpp_Objects)
 
 $(Signed_Enclave_Name): $(Enclave_Name)
 	@$(SGX_ENCLAVE_SIGNER) sign -key Enclave/enclave_config/enclave_hsm_private.pem -enclave $(Enclave_Name) -out $@ -config $(Enclave_Config_File)
-	@echo "SIGN =>  $@"
+	@echo "SIGN =>  $@"
```
Comment (yang8621) [inline core/Makefile:167]: Discard this change: 100644 → 100755

File: docker/Dockerfile [hunk 4]
```diff
@@ -61,8 +61,26 @@ COPY bin/ehsm-core/libenclave-ehsm-core.signed.so /home/ehsm/ehsm-core/libenclav
 #    && cd ehsm \
 #    && make
 
+# node-service
+RUN mkdir -p /home/ehsm/ehsm-service
+WORKDIR /home/ehsm/ehsm-service
+
+RUN wget https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.xz \
```
Comment (yang8621) [inline docker/Dockerfile:68]: Does it mean "apt install nodejs" is not required?

File: docker/Dockerfile [hunk 5]
```diff
@@ -61,8 +61,26 @@ COPY bin/ehsm-core/libenclave-ehsm-core.signed.so /home/ehsm/ehsm-core/libenclav
...
+    && tar xf  node-v16.13.0-linux-x64.tar.xz \
+    && mv node-v16.13.0-linux-x64/ /usr/local/nodejs \
+    && ln -s /usr/local/nodejs/bin/node /usr/local/bin \
+    && ln -s /usr/local/nodejs/bin/npm /usr/local/bin
+    # && npm config set registry https://registry.npm.taobao.org/
+
+COPY /ehsm-service /home/ehsm/ehsm-service
+
+RUN npm install
```
Comment (yang8621) [inline docker/Dockerfile:77]: is "apt install npm" required?

File: docker/Dockerfile [hunk 6]
```diff
@@ -61,8 +61,26 @@ COPY bin/ehsm-core/libenclave-ehsm-core.signed.so /home/ehsm/ehsm-core/libenclav
 #    && cd ehsm \
 #    && make
```
Comment (yang8621) [inline docker/Dockerfile:63]: I would suggest COPY the libehsmnapi.so and libenclave-ehsm-core.signed.so in Dockerfile but not commit them into this patch.

File: ehsm-service/index.js (hunk 1 above)
Comment (Liang1XZhao) [reply ehsm-service/index.js:57]: ok

**Pull request closed without merging**